zebrad/src/commands/start.rs

```diff
@@ -39,6 +39,7 @@ pub struct StartCmd {
 impl StartCmd {
     async fn start(&self) -> Result<(), Report> {
         info!(?self, "starting to connect to the network");
+        tokio::spawn(on_ctrl_c());
```

Comment:
Issue: The flame graph is initialised when `ZebradApp::command_is_server(&command)`, but only dropped for the start command.

Suggestion: consider also writing the flame graph when the process exits for other reasons. For example, hangup, pipe, and terminate are common unix signals:
https://docs.rs/tokio/0.2.22/tokio/signal/unix/struct.SignalKind.html#method.hangup

You might also want to write the graph on quit, before the process dumps core. (It would be surprising for a graph to be written on some common signals, but not others.)

Reply:
Since the flamegraph is mostly meant as a performance profiling tool and not as a debugging or otherwise normal usage tool I don't think we need to harden it for all signals or defend against unexpected behaviour, the only people who are using this will likely be people familiar enough with the project to figure out why they didn't get a flame graph for their specific exit mode. I'd prefer to keep the implementation as simple as possible and add new support for other signals and termination modes as needed rather than proactively.